src/transactions/transactions.util.ts

```diff
@@ -53,11 +69,35 @@ export async function processTransaction<
 >(
   method: Method<MethodArgs, ReturnType, TransformedReturnType>,
   args: MethodArgs,
-  opts: ProcedureOpts
+  opts: ProcedureOpts,
+  dryRun = false
 ): Promise<TransactionResult<TransformedReturnType>> {
   try {
     const procedure = await prepareProcedure(method, args, opts);
-    const result = await procedure.run();
+    const {
+      fees,
+      payingAccountData: { balance, type, account },
+    } = await procedure.getTotalFees();
+    const supportsSubsidy = await procedure.supportsSubsidy();
```

Comment:
I am not sure on a good pattern for this and it will take some more thinking, but:

I think this should be done in parallel with the actual running of the transaction so we aren't adding extra latency to every transaction.